pulser/sequence.py

```diff
@@ -332,7 +356,9 @@ def declare_variable(self, name, size=1, dtype=float):
         return var
 
     @_store
-    def add(self, pulse, channel, protocol='min-delay'):
+    def add(self, pulse: Union[Pulse, Parametrized],
+            channel: Union[str, Parametrized],
+            protocol: Union[str, Parametrized] = 'min-delay') -> None:
```

Comment:
Since when `protocol` is a `str`, it can only be `'min-delay', 'no-delay', 'wait-for-all'`, I think it would be nice to use a [`Literal`](https://docs.python.org/3/library/typing.html#typing.Literal) instead of `str`.

I would do an alias for valid protocols, like is done with `MODE` in the link above.